contracts/connectors/loantoken/LoanTokenLogicBeacon.sol

```diff
@@ -64,13 +72,46 @@ contract LoanTokenLogicBeacon is PausableOz {
 		(bytes4[] memory functionSignatureList, bytes32 moduleName) =
 			ILoanTokenLogicModules(loanTokenModuleAddress).getListFunctionSignatures();
 
+		/// register / update the module function signature address implementation
```

Comment:
It might be cleaner or more efficient to utilize list of signatures received directly from the old module, smth like this:
```
address prevVersion = moduleUpgradeLog[moduleName][activeModuleIndex[moduleName]];
(bytes4[] memory oldSigList, bytes32  oldName) = ILoanTokenLogicModules(prevVersion ).getListFunctionSignatures();
assert(moduleName == oldName);
```
and then performing similar add/remove operations but with less storage overhead.
(I didn't dive into it too deep)

Anyway, this code shouldn't be called too often, so correctness is more important than high efficiency

Reply:
We need EnumerableBytes4Set in this case so that it can be easier for the comparison inside the loop. Meanwhile the getListFunctionSignatures() returns array.
So in this case I prefer to use the enumerate() to get the active / previous function signature list.